### PR TITLE
neovide 0.9.0 (new formula)

### PR DESCRIPTION
**Pull request by peeweep**
Signed-off-by: peeweep <peeweep@0x0.ee>

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/neovide.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    depends_on "libxcb"
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+    bin.install "target/release/neovide"
+  end
+
+  test do
+    system bin/"neovide", "--remote-tcp=localhost:6666"
```
Comment (carlocab) [inline Formula/neovide.rb:23]: Don't we want to actually start an `nvim` server for this to actually test something? I'm surprised trying to connect to a non-existent server doesn't error, actually.

**Comment by peeweep** [comment]
I'm at a loss on how to write a test block right now, I'm recently moving and I don't have much time to read neovide's code. If no one can help modify the test block, it might be better to close this issue.

**Comment by carlocab** [comment]
I've had a look at improving the `test` block. Let's give this a try.

File: Formula/neovide.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  test do
+    puts shell_output("#{bin}/neovide --help")
+    assert_match version.to_s, shell_output("#{bin}/neovide --version")
+
+    test_server = "localhost:#{free_port}"
+    nvim_pid = spawn "nvim", "--headless", "--listen", test_server
+    neovide_pid = spawn bin/"neovide", "--nofork", "--remote-tcp=#{test_server}"
+    system "nvim", "--server", test_server, "--remote-send", ":q<CR>"
```
Comment (carlocab) [inline Formula/neovide.rb:29]:
```suggestion
    neovide_pid = spawn bin/"neovide", "--nofork", "--remote-tcp=#{test_server}"
    sleep 10
    system "nvim", "--server", test_server, "--remote-send", ":q<CR>"
```

File: Formula/neovide.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    depends_on "libxcb"
+  end
+
+  def install
+    system "cargo", "install", *std_cargo_args
+    bin.install "target/release/neovide"
+  end
+
+  test do
+    puts shell_output("#{bin}/neovide --help")
```
Comment (carlocab) [inline Formula/neovide.rb:23]:
```suggestion
```

File: Formula/neovide.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    system "cargo", "install", *std_cargo_args
+    bin.install "target/release/neovide"
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/neovide --version")
+
+    test_server = "localhost:#{free_port}"
+    nvim_pid = spawn "nvim", "--headless", "--listen", test_server
+    neovide_pid = spawn bin/"neovide", "--nofork", "--remote-tcp=#{test_server}"
```
Comment (carlocab) [inline Formula/neovide.rb:27]:
```suggestion
    nvim_pid = spawn "nvim", "--headless", "--listen", test_server
    sleep 10
    neovide_pid = spawn bin/"neovide", "--nofork", "--remote-tcp=#{test_server}"
```

**Comment by carlocab** [comment]
Ok, test block is a bit better. But now the Linux build is broken. Care to take a look, @peeweep?

**Comment by peeweep** [comment]
Thanks for complete the test block, I will bump the version to 0.9.

**Review by carlocab — DISMISSED** [review]
Thanks, @peeweep.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**